lib/src/main/java/com/diffplug/spotless/json/JsonFormatterStep.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright 2021 DiffPlug
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.diffplug.spotless.json;
+
+import java.io.IOException;
+import java.io.Serializable;
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Method;
+import java.util.Objects;
+
+import com.diffplug.spotless.FormatterFunc;
+import com.diffplug.spotless.FormatterStep;
+import com.diffplug.spotless.JarState;
+import com.diffplug.spotless.Provisioner;
+
+public final class JsonFormatterStep {
+	private static final String MAVEN_COORDINATE = "org.json:json:";
+	private static final String DEFAULT_VERSION = "20210307";
+	private static final int INDENT = 4;
```

Comment:
:question: should this be configurable?

Reply:
There is *guaranteed* to be confusion with the `indentWithSpaces` and `indentWithTabs` rules, unfortunately, but I don't think there's anything we can do about that here.

There are lots of configurable things I can imagine. Indentation, but also field ordering, perhaps putting small objects on a single line, etc. We don't need to add everything right now, but it probably does make sense to at least start with `indentWithSpaces('4')` vs `indentWithTabs()`. It's gonna be a little confusing until #794 happens, but that shouldn't stop us from doing the right thing here.

Reply:
That's fair enough. To confirm, we should make this configurable at this point? I'll align with the `indentWithSpaces` naming if so

Reply:
Yes, we should make it configurable, and I would make the methods `indentWithSpaces(int num)` and `indentWithTabs()` (no reason to indent with multiple tabs). That's a tiny bit different than the existing methods, but I think it's an improvement.

Reply:
Sure thing, I didn't initially add tabs as it wasn't out of the box with the library but makes sense to add it!

Reply:
Sorry, I'm not quite understanding how to take the configuration from `indentWithSpaces(int num)` and `indentWithTabs()` and pass them to the `JsonExtension` / `JsonFormatterStep` - is there anything you can point me to that does it similarly?

Reply:
I would do it like this:

- add a field to JsonFormatterStep, `Integer indent` where null codes for tab
- `indentWithSpaces/Tabs` are just methods for the gradle configuration, e.g.

https://github.com/diffplug/spotless/blob/c09f3cb182bed3c3fa0c92d155bbe47756093312/plugin-gradle/src/main/java/com/diffplug/gradle/spotless/KotlinExtension.java#L142-L170

Reply:
Thanks I think that gives me a good amount to go off 👍🏽

Reply:
I'm not quite sure what's going wrong, but every time I try to add something after `target`, I end up with i.e.

```
* What went wrong:
Could not determine the dependencies of task ':spotlessApply'.
> Could not create task ':spotlessJsonApply'.
   > Could not create task ':spotlessJson'.
      > Cannot invoke method indentWithSpaces() on null object
```

I'm not sure where I'm going wrong - are you able to nudge me in the right direction?

This happens when I've got the custom methods for `indentWithSpaces` / `indentWithTabs` and when I've left it to be inherited from `FormatExtension` :thinking:

Reply:
The problem is confusing a `FormatExtension` subclass with a formatter configuration. Note that `DiktatFormatExtension` has no superclass, and its whole goal is just to configure that one step. See comments below for more info.